### ISSUE TITLE
multiplayer -> load game dialog should correctly handle replays (GNA #21929)

### ISSUE THREAD

**Issue by wesnoth-bugs** [wesnoth/wesnoth#1382]
**[Original submission](https://gna.org/bugs/index.php?21929)** by involution on 2014-04-16

right now it fails with "this scenario has no id", and if that check is disabled, with "this scenario has no sides". i would think that we would be able to detect a replay and launch the replay_controller, similarly to how we do for title screen -> load game.

(Reproduced on any)
Release: 1.12,1.13-dev
Priority: 5 - Normal
Severity: 3 - Normal

**Comment by wesnoth-bugs** [comment]
**Modified on 2014-04-16**

**involution wrote:**

> gfgtdf points out that this is most likely the same as the unsolved portion of this bug: https://gna.org/bugs/?20895

**Comment by wesnoth-bugs** [comment]
**Modified on 2014-04-18**

**anonymous wrote:**

> even when the "this scenario has no id" bug is fixed what you expect to happen whe you click waetch replay in mp?
> 
> We currently have different codepaths for watching replays (replay_controller) and mp games (playmp_controller). So being able to watch a replay together with others via network would require completeley new code and i would consider it rather a feature than a bug that this is not possible. Since you can still watch mp replays in the normal load game dilaog i'd be ok if we just disable tht button in mp load game dialog, maybe we should make that warning message disappear in this case though.

**Comment by wesnoth-bugs** [comment]
**Modified on 2014-04-18**

**involution wrote:**

> "even when the "this scenario has no id" bug is fixed what you expect to happen whe you click waetch replay in mp? "
> 
> 
> 
> I'd expect the replay controller to be launched, similarly to to the SP -> Load game option. Ideally this would happen by a code path that would definitely not do SP specific things, and we wouldn't have the warning "this is an mp scenario. some things may not work properly" etc.
> 
> 
> 
> "So being able to watch a replay together with others via network would require completeley new code and i would consider it rather a feature than a bug that this is not possible." 
> 
> 
> 
> Yes ofc I agree, in fact I think that is actually a feature request already.
> 
> 
> 
> "Since you can still watch mp replays in the normal load game dilaog i'd be ok if we just disable tht button in mp load game dialog, maybe we should make that warning message disappear in this case though."
> 
> 
> 
> No that would be a mistake. The replay check box in the load game dialog serves a useful purpose, it determine whether we will load from snapshot or from replay_start and replay.

**Comment by wesnoth-bugs** [comment]
**Modified on 2015-07-24**

**gfgtdf wrote:**

> > No that would be a mistake. The replay check box in the load game dialog serves a useful purpose, it determine whether we will load from snapshot or from replay_start and replay.
> 
> 
> 
> Yes, that proposaal was just about the multiplayer loadgame dialog (because it doesnt work anyway there), the sp load dialog shoudl stay unchanged.

**gfgtdf changed category:** Bug -> Feature Request

**Comment by wesnoth-bugs** [comment]
**Modified on 2016-01-10**

**vultraz wrote:**

> This appears to be fixed. Should I close?

**Comment by wesnoth-bugs** [comment]
**Modified on 2016-01-10**

**vultraz wrote:**

> Actually, nvm, I misunderstood the bug (to be sure - this does mean the Load Game button in the Create Game screen in MP, right?)

**Comment by Wedge009** [comment]
@Vultraz Yes, I understand this to be the load game in MP lobby. As of 1.14.7 there is at least a warning message that replays cannot be loaded in MP mode.
